modules/videoio/src/cap_dshow.cpp

```diff
@@ -2946,6 +2946,22 @@ int videoInput::start(int deviceID, videoDevice *VD){
     DebugPrintOut("SETUP: Device is setup and ready to capture.\n\n");
     VD->readyToCapture = true;
 
+    // check for optional saving the direct show graph to a file
+    const char* graph_filename = getenv("OPENCV_DSHOW_SAVEGRAPH_FILENAME");
+    if (graph_filename) {
+        WCHAR* wfilename = new WCHAR[strlen(graph_filename) + 1];
+        mbstowcs(wfilename, graph_filename, strlen(graph_filename) + 1);
```

Comment:
return value of `mbstowcs()` should be checked (asserted)

Reply:
Done